examples/slash/slash.go

```diff
@@ -11,24 +11,36 @@ import (
 
 func main() {
 	var (
-		verificationToken string
+		signingSecret string
 	)
 
-	flag.StringVar(&verificationToken, "token", "YOUR_VERIFICATION_TOKEN_HERE", "Your Slash Verification Token")
+	flag.StringVar(&signingSecret, "secret", "YOUR_SIGNING_SECRET_HERE", "Your Slack app's signing secret")
 	flag.Parse()
 
 	http.HandleFunc("/slash", func(w http.ResponseWriter, r *http.Request) {
-		s, err := slack.SlashCommandParse(r)
+
+		verifier, err := slack.NewSecretsVerifier(r.Header, signingSecret)
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 
-		if !s.ValidateToken(verificationToken) {
+		_, err = verifier.WriteBody(r)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		if err = verifier.Ensure(); err != nil {
 			w.WriteHeader(http.StatusUnauthorized)
 			return
 		}
 
+		s, err := slack.SlashCommandParse(r)
```

Comment:
this also will likely not work the way you expect. as you've already read from r.

Reply:
this does not work with io.Copy. i have tested locally and on a remote server taking real slack commands and the new method correctly verifies the request and parses the command from the request afterwards.